hotdoc/core/database.py

```diff
@@ -114,8 +120,20 @@ def get_or_create_symbol(self, type_, **kwargs):
 
         filename = kwargs.get('filename')
         if filename:
+            filename = os.path.abspath(filename)
             kwargs['filename'] = os.path.abspath(filename)
 
+        prev_sym = self.__symbols.get(unique_name)
+        if prev_sym:
+            # pylint: disable=unidiomatic-typecheck
+            if (prev_sym.filename and filename and prev_sym.filename != filename) or (
+                    prev_sym.project_name != kwargs.get('project_name')) or (
+                        type(prev_sym) != type_):
+                warn('symbol-redefined', "%s(unique_name=%s, filename=%s, project=%s)"
+                     " has already been defined: %s" % (type_.__name__, unique_name, filename,
+                                                        kwargs.get('project_name'), prev_sym))
+                return None
```

Comment:
All code calling on this function should be updated to account for the fact that it can now return None